### ISSUE TITLE
ZeroPoleWeight is not necessarily real

### ISSUE THREAD

**Issue by krivenko** [aeantipov/pomerol#26]
First of all, I would like to thank Junya @j-otsuki for his major contribution! The ability to compute susceptibilities has certainly been a much requested feature.

Today I discovered a compilation failure seen only in the `POMEROL_COMPLEX_MATRIX_ELEMENT=ON` mode.
```
/home/igor/Physics/pomerol/pomerol.git/src/pomerol/SusceptibilityPart.cpp: In member function ‘void Pomerol::SusceptibilityPart::compute()’:
/home/igor/Physics/pomerol/pomerol.git/src/pomerol/SusceptibilityPart.cpp:68:36: error: no match for ‘operator+=’ (operand types are ‘Pomerol::RealType’ {aka ‘double’} and ‘std::complex<double>’)
                     ZeroPoleWeight += Ainner.value() * Binner.value() * DMpartOuter.getWeight(index1);
                     ~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

I guess putting an `std::real()` around the RHS would make the error go away, but the problem is actually deeper than that. `ZeroPoleWeight` is declared to be real, but this property is guaranteed only when operator `B` is the Hermitian conjugate of `A` (if I am not mistaken).
Therefore, we should either declare `ZeroPoleWeight` as `MelemType`, or check that `B = A^+` in the constructor of `Susceptibility`.

**Comment by aeantipov** [comment]
They don't have to be the conjugates right? In principle A and B could be operators on different sites/orbitals. So probably MelemType is the way to go.

**Comment by j-otsuki** [comment]
Thank you Igor for pointing it out. I think we should declare ZeroPoleWeight as MelemType, as Andrey says. Igor, can you fix it and check if the compilation failure is resolved?

**Comment by krivenko** [comment]
@aeantipov @j-otsuki Thanks for your responses! Fixed in 7a45b6a.
